# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGAnimatedLength.cpp
# Commit: 4cca882e069d
# Full Hash: 4cca882e069d67c3a9a9064ac603b0dca23a5b39
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 21:17:53
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 2f48a6566514 -r 4cca882e069d dom/svg/SVGAnimatedLength.cpp
--- a/dom/svg/SVGAnimatedLength.cpp	Tue Sep 12 11:45:26 2023 +0000
+++ b/dom/svg/SVGAnimatedLength.cpp	Tue Sep 12 11:45:27 2023 +0000
@@ -178,6 +178,11 @@
   return GetEmLength(aType);
 }
 
+CSSSize UserSpaceMetrics::GetCSSViewportSizeFromContext(
+    const nsPresContext* aContext) {
+  return CSSPixel::FromAppUnits(aContext->GetSizeForViewportUnits());
+}
+
 SVGElementMetrics::SVGElementMetrics(const SVGElement* aSVGElement,
                                      const SVGViewportElement* aCtx)
     : mSVGElement(aSVGElement), mCtx(aCtx) {}
@@ -210,6 +215,17 @@
   return FixAxisLength(mCtx->GetLength(aCtxType));
 }
 
+CSSSize SVGElementMetrics::GetCSSViewportSize() const {
+  if (!mSVGElement) {
+    return {0.0f, 0.0f};
+  }
+  nsPresContext* context = nsContentUtils::GetContextForContent(mSVGElement);
+  if (!context) {
+    return {0.0f, 0.0f};
+  }
+  return GetCSSViewportSizeFromContext(context);
+}
+
 bool SVGElementMetrics::EnsureCtx() const {
   if (!mCtx && mSVGElement) {
     mCtx = mSVGElement->GetCtx();
@@ -279,6 +295,10 @@
   return SVGIntegrationUtils::GetSVGCoordContextForNonSVGFrame(mFrame);
 }
 
+CSSSize NonSVGFrameUserSpaceMetrics::GetCSSViewportSize() const {
+  return GetCSSViewportSizeFromContext(mFrame->PresContext());
+}
+
 float UserSpaceMetricsWithSize::GetAxisLength(uint8_t aCtxType) const {
   gfx::Size size = GetSize();
   float length;